impeller/entity/entity_unittests.cc

```diff
@@ -1736,5 +1744,159 @@ TEST_P(EntityTest, SrgbToLinearFilter) {
+          {BlendMode::kSourceOut,
+           [](const Color& src, const Color& dst) {
+             return Color(
+                 (1 - dst.alpha) * src.red, (1 - dst.alpha) * src.green,
+                 (1 - dst.alpha) * src.blue, (1 - dst.alpha) * src.alpha);
+           }},
+          {BlendMode::kDestinationOut,
+           [](const Color& src, const Color& dst) {
+             return Color(
+                 (1 - src.alpha) * dst.red, (1 - src.alpha) * dst.green,
+                 (1 - src.alpha) * dst.blue, (1 - src.alpha) * dst.alpha);
+           }},
+          {BlendMode::kSourceATop,
+           [](const Color& src, const Color& dst) {
+             return Color(dst.alpha * src.red + (1 - src.alpha) * dst.red,
+                          dst.alpha * src.green + (1 - src.alpha) * dst.green,
+                          dst.alpha * src.blue + (1 - src.alpha) * dst.blue,
+                          dst.alpha);
+           }},
+          {BlendMode::kDestinationATop,
+           [](const Color& src, const Color& dst) {
+             return Color(src.alpha * dst.red + (1 - dst.alpha) * src.red,
+                          src.alpha * dst.green + (1 - dst.alpha) * src.green,
+                          src.alpha * dst.blue + (1 - dst.alpha) * src.blue,
+                          src.alpha);
+           }},
+          {BlendMode::kXor,
+           [](const Color& src, const Color& dst) {
+             return Color(
+                 (1 - src.alpha) * dst.red + (1 - dst.alpha) * src.red,
+                 (1 - src.alpha) * dst.green + (1 - dst.alpha) * src.green,
+                 (1 - src.alpha) * dst.blue + (1 - dst.alpha) * src.blue,
+                 (1 - src.alpha) * dst.alpha + (1 - dst.alpha) * src.alpha);
+           }},
+          {BlendMode::kPlus,
+           [](const Color& src, const Color& dst) {
+             return Color(std::max(0.f, std::min(src.red + dst.red, 1.f)),
+                          std::max(0.f, std::min(src.green + dst.green, 1.f)),
+                          std::max(0.f, std::min(src.blue + dst.blue, 1.f)),
+                          std::max(0.f, std::min(src.alpha + dst.alpha, 1.f)));
+           }},
+          {BlendMode::kModulate, [](const Color& src, const Color& dst) {
+             return Color(src.red * dst.red, src.green * dst.green,
+                          src.blue * dst.blue, src.alpha * dst.alpha);
+           }}};
+
+  auto check_func = [blend_modes, &blend_color_map](Color& src, Color& dst) {
+    FML_LOG(ERROR) << "Source " << src << " Dst " << dst;
+    for (unsigned i = 0; i < blend_modes.size(); ++i) {
+      auto blend_color = Color::BlendColor(src, dst, blend_modes[i]);
+      if (blend_color_map.find(blend_modes[i]) != blend_color_map.end()) {
+        auto expected_color = blend_color_map[blend_modes[i]](src, dst);
+        ASSERT_EQ(blend_color, expected_color);
+      }
+    }
+  };
+
+  {
+    Color src = {1, 0, 0, 0.5};
+    Color dst = {1, 0, 1, 1};
+    check_func(src, dst);
```

Comment:
What I mean is, you should explicitly assert what you expect the color to be in this assertion instead of computing it on the fly